lib/rubocop/cop/style/empty_class.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Prefer a single-line format for class definitions with no body.
+
+      # @example EnforcedStyle: compact (default)
+      #   # bad
+      #   class FooError < StandardError
+      #   end
+      #
+      #   # bad
+      #   FooError = Class.new(StandardError)
+      #
+      #   # good
+      #   class FooError < StandardError; end
+      #
+      # @example EnforcedStyle: compact_constant
+      #   # bad
+      #   class FooError < StandardError
+      #   end
+      #
+      #   # bad
+      #   class FooError < StandardError; end
+      #
+      #   # good
+      #   FooError = Class.new(StandardError)
+      #
+      class EmptyClass < Cop
+        MSG = 'Use a single-line format for class definitions with no body.'
+
+        def on_class(node)
+          return if !node.children.last.nil? || node.single_line?
```

Comment:
I guess we should also add an offence if `node.single_line?` in case the enforced style is `compact_constant`.